src/figuresClasses.ts

```diff
@@ -1,19 +1,97 @@
-export interface Figure {
+type Shape =
+  'triangle' |
+  'circle' |
+  'rectangle';
+
+type Color =
+  'red' |
+  'green' |
+  'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea: () => {};
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape;
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    this.shape = 'triangle';
 
-export class Circle {
+    if (a <= 0
+      || b <= 0
+      || c <= 0
+      || a + b <= c
+      || a + c <= b
+      || b + c <= a
+    ) {
+      throw new Error('invalid sides');
+    }
+  }
 
+  getArea(): number {
+    const s: number = (this.a + this.b + this.c) / 2;
+
+    return Math.floor(Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c))
+     * 100) / 100;
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  shape: Shape;
+
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    this.shape = 'circle';
+
+    if (radius <= 0) {
+      throw new Error('invalid radius');
+    }
+  }
 
+  getArea(): number {
+    return Math.floor((Math.PI * this.radius ** 2) * 100) / 100;
+  }
 }
 
-export function getInfo(figure) {
+export class Rectangle implements Figure {
+  shape: Shape;
+
+  color;
+
+  width;
+
+  height;
+
+  constructor(
+    color: Color,
+    width: number,
+    height: number,
+  ) {
+    this.shape = 'rectangle';
+    this.color = color;
+    this.width = width;
```

Comment:
You can do it simpler
```suggestion
  constructor(
    public color: Color,
    public width: number,
    public height: number,
  ) {
    this.shape = 'rectangle';
```